test/logflare/backends/postgres_adaptor_test.exs

```diff
@@ -197,14 +197,36 @@ defmodule Logflare.Backends.Adaptor.PostgresAdaptorTest do
   end
 
   test "bug: cast_config/1 and validate_config/1 postgresql url variations" do
-    assert %Ecto.Changeset{valid?: true} = %{url: "postgresql://localhost:5432"} |> PostgresAdaptor.cast_config() |> PostgresAdaptor.validate_config()
-    assert %Ecto.Changeset{valid?: true} = %{url: "postgres://localhost:5432"} |> PostgresAdaptor.cast_config() |> PostgresAdaptor.validate_config()
+    assert %Ecto.Changeset{valid?: true} =
+             %{url: "postgresql://localhost:5432"}
+             |> PostgresAdaptor.cast_config()
+             |> PostgresAdaptor.validate_config()
+
+    assert %Ecto.Changeset{valid?: true} =
+             %{url: "postgres://localhost:5432"}
+             |> PostgresAdaptor.cast_config()
+             |> PostgresAdaptor.validate_config()
 
     # invalid connection strings
-    assert %Ecto.Changeset{valid?: false} = %{url: "://localhost:5432"} |> PostgresAdaptor.cast_config() |> PostgresAdaptor.validate_config()
-    assert %Ecto.Changeset{valid?: false} = %{url: "//localhost:5432"} |> PostgresAdaptor.cast_config() |> PostgresAdaptor.validate_config()
-    assert %Ecto.Changeset{valid?: false} = %{url: "/localhost:5432"} |> PostgresAdaptor.cast_config() |> PostgresAdaptor.validate_config()
-    assert %Ecto.Changeset{valid?: false} = %{url: "postgres//localhost:5432"} |> PostgresAdaptor.cast_config() |> PostgresAdaptor.validate_config()
+    assert %Ecto.Changeset{valid?: false} =
+             %{url: "://localhost:5432"}
+             |> PostgresAdaptor.cast_config()
+             |> PostgresAdaptor.validate_config()
+
+    assert %Ecto.Changeset{valid?: false} =
+             %{url: "//localhost:5432"}
+             |> PostgresAdaptor.cast_config()
+             |> PostgresAdaptor.validate_config()
+
+    assert %Ecto.Changeset{valid?: false} =
+             %{url: "/localhost:5432"}
+             |> PostgresAdaptor.cast_config()
+             |> PostgresAdaptor.validate_config()
+
+    assert %Ecto.Changeset{valid?: false} =
+             %{url: "postgres//localhost:5432"}
+             |> PostgresAdaptor.cast_config()
+             |> PostgresAdaptor.validate_config()
   end
```

Comment:
I would create function like:

```elixir
defp cast_config(attrs) do
  attrs
  |> Map.new()
  |> PostgresAdaptor.cast_config()
  |> PostgresAdaptor.validate_config()
end
```

And then in tests I would do:

```elixir
refute cast_config(url: "postgresql//localhost:5432").valid?
```

Also I would make separate test for each of the cases.